src/notification/NotificationController.ts

```diff
@@ -0,0 +1,86 @@
+import BaseController, { BaseConfig, BaseState } from '../BaseController';
+
+interface viewedNotification {
+[id: number]: boolean;
+}
+
+export interface Notification{
+    id: number;
+    title: string;
+    description: string;
+    date: string;
+    image?: string;
+    actionText: string;
+    isShown?: boolean;
+}
+
+// NotitificationConfig will hold the notifications from JSON file read from `metamask-extension`
+export interface NotificationConfig extends BaseConfig{
+    notificationsFromFile: { [whatsnew: string]: Notification[]};
+}
+
+// Notification state will hold all the seen and unseen notifications
+// that are still active
+export interface NotificationState extends BaseState{
+    notifications: Notification[];
+}
+
+const defaultState = { notifications: [] };
+
+/**
+ * Controller for managing the notifications for new features introducesd into the applications
+ */
+export class NotificationController extends BaseController<NotificationConfig, NotificationState> {
+
+    constructor(config: NotificationConfig, state?: NotificationState) {
+        const { notificationsFromFile } = config;
+        super(config, state || defaultState);
+        this.initialize();
+        this.addNotifications(notificationsFromFile.whatsnew);
+    }
+
+    /**
+     * Compares the notifications in teh states with the notifications from the files to check if there is any new notitifcations/announcements
+     * if yes, the new notification will be added to the state with a flag indicating that the notification is not seen by the user.
+     * @param filedNotifications
+     */
+    private addNotifications(filedNotifications: Notification[]): void{
+        const stateNotifications = this.state.notifications;
+        let exists: boolean;
+        if (this.state.notifications.length > 0) {
```

Comment:
This condition doesn't seem necessary. Now the logic for marking a notification as unseen is duplicated in two places. This would work the same way if this condition and the `else` block was deleted.